src/services/chains/solana/transactionService.ts

```diff
@@ -62,10 +79,55 @@ export const getSolanaTransactionDetailForSolanaTransfer = async (
   }
 };
 
-function getTransactionDetailForTokenTransfer(
-  input: TransactionDetailInput,
-): Promise<NetworkTransactionInfo> {
-  throw new Error('Not Implemented');
+async function getTransactionDetailForSplTokenTransfer(
+  params: TransactionDetailInput,
+): Promise<NetworkTransactionInfo | null> {
+  try {
+    const SPL_TOKEN_TRANSFER_INSTRUCTION_TYPE = 'spl-token';
+    const TRANSFER_CHECK_TYPE = 'transferChecked';
+
+    const result = await getSolanaWebProvider(
+      params.networkId,
+    ).getParsedTransaction(params.txHash);
+    const token = await findTokenByNetworkAndSymbol(
+      params.networkId,
+      params.symbol,
+    );
+
+    const data = result?.transaction?.message?.instructions?.find(
+      instruction =>
+        instruction?.program === SPL_TOKEN_TRANSFER_INSTRUCTION_TYPE &&
+        instruction?.parsed?.type === TRANSFER_CHECK_TYPE &&
+        instruction?.parsed?.info.authority === params.fromAddress &&
+        instruction?.parsed?.info.mint === token?.address,
+    );
+    const toAddressBalance = result?.meta?.postTokenBalances?.find(
+      balance =>
+        balance.owner === params.toAddress && balance.mint === token?.address,
+    );
+    if (!data || !toAddressBalance) {
+      return null;
+    }
+    const parsedData = data as ParsedInstruction;
+
+    const txInfo = parsedData.parsed.info;
+    if (!txInfo) {
+      return null;
+    }
+    return {
+      from: txInfo.authority,
+
+      // we already check toAddressBalance.owner === params.toAddress
+      to: toAddressBalance?.owner,
+      amount: txInfo.tokenAmount?.uiAmount,
```

Comment:
The example I have doesn't hold any value in parsed.info.tokenAmount.uiAmount!
The safer approach would be get receivers balance in preTokenBalances and postTokenBalances, then use their diff here.

Reply:
I got my example by running this
```ts
 const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
  const txInfo = await connection.getParsedTransaction(
    "3m6f1g2YK6jtbfVfuYsfDbhVzNAqozF8JJyjp1VuFDduecojqeCVK4htKnLTSk3qBwSqYUvgLpBTVpeLJRvNmeTg"
  );
  console.log(JSON.stringify(txInfo, null, 2));
  ```

Reply:
let me write a test case for your example, to see what happens

Reply:
Thanks I'm working on it